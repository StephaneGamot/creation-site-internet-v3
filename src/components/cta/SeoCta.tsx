"use client"

import ContactButton from "../Buttons/ContactButton"
import { useTranslations } from "next-intl"

export default function SeoCta() {
  const t = useTranslations("seoCta");

  return (
    <section className="text-center pb-0">
      <h2 className="text-3xl font-title font-semibold mb-4">
        {t("title")}
      </h2>
      <p className="text-lg text-gray-700 mb-6">
        {t("text")}
      </p>
      <ContactButton />
    </section>
  );
}
