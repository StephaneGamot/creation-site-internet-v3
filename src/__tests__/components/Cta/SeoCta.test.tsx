import React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import SeoCta from "@/components/cta/SeoCta";

// ✅ Mock des traductions
jest.mock("next-intl", () => ({
  useTranslations: () => {
    return (key: string) => {
      if (key === "title") return "Besoin d’un SEO stratégique ?";
      if (key === "text") return "Votre visibilité mérite mieux que des mots-clés aléatoires. Parlons stratégie.";
      if (key === "aria") return "Aller à la page de contact";
      if (key === "label") return "Me contacter";
      return "";
    };
  },
  useLocale: () => "fr",
}));

// ✅ Mock du bouton
jest.mock("@/components/Buttons/ContactButton", () => () => (
  <a href="/fr/contact" role="button">Me contacter</a>
));

describe("SeoCta", () => {
  it("affiche le titre et le texte", () => {
    render(<SeoCta />);
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Besoin d’un SEO stratégique ?");
    expect(screen.getByText("Votre visibilité mérite mieux que des mots-clés aléatoires. Parlons stratégie.")).toBeInTheDocument();
  });

  it("affiche le bouton Me contacter", () => {
    render(<SeoCta />);
    const button = screen.getByRole("button", { name: "Me contacter" });
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute("href", "/fr/contact");
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<SeoCta />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const { asFragment } = render(<SeoCta />);
    expect(asFragment()).toMatchSnapshot();
  });
});
