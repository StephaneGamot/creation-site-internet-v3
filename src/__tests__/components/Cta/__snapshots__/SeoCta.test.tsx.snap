// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SeoCta correspond au snapshot 1`] = `
<DocumentFragment>
  <section
    class="text-center pb-0"
  >
    <h2
      class="text-3xl font-title font-semibold mb-4"
    >
      Besoin d’un SEO stratégique ?
    </h2>
    <p
      class="text-lg text-gray-700 mb-6"
    >
      Votre visibilité mérite mieux que des mots-clés aléatoires. Parlons stratégie.
    </p>
    <a
      href="/fr/contact"
      role="button"
    >
      Me contacter
    </a>
  </section>
</DocumentFragment>
`;
